### ISSUE TITLE
get_resource_limits() in privileged api should be accessed by everyone

### ISSUE THREAD

**Issue by qimiaoguo** [EOSIO/eos#5319]
Now only the privileged account can access the privileged api, this is true.
However, I think the get_resource_limits should be take out of it because it is just a get-action.

Two reasons:
1. My contract can know how much ram or net or cpu has been used when an inline action is executed, then it can decide if to accept it or not.
2. When I transfer token to others, if others' account take your ram, then you can monitor it and let the transfer fail.

Now we can only get account usage with the RPC, but not in the contract, is it right?

**Comment by taokayan** [comment]
agree with this. But it's not an easy change, as it might result in hard-fork or non-replayable chain.

**Comment by qimiaoguo** [comment]
@taokayan Why can it result in a hard-fork? It is a new action that  no contract have used before.

**Comment by wanderingbort** [comment]
@qimiaoguo the issue that creates a hardfork is the change in access to the `get_resource_limits` intrinsic.  consider a deployed un-privileged contract that uses this intrinsic in an action called `forkchain` which does nothing but call this intrinsic.

Any validator/producer that is running a version before this change would consider any transaction that includes `forkchain` to be a violation of the protocol as it calls an intrinsic that the software considers to be "privileged-only"

Any validator/producer that is running a version after this change would consider any transaction that includes `forchain` valid.

So, any block that includes said transaction is a hard-fork point where any old client will fork out of the network and only new clients will be able to proceed.

**Comment by wanderingbort** [comment]
additionally `get_resource_limits` provides limits, not usage.  So it cannot detect when an account has used your accounts resources in an unexpected way.

**Comment by liamcurry** [comment]
Agreed that getters like `get_resource_limits` and `get_blockchain_parameters_packed` should be available to everyone.
